### PR TITLE
Removing type_size trait, replacing it with special archive type

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

**Comment by sithhell** [comment]
How are zero copy chunks handled? Do potential zero copy arguments
contribute to the size?

File: tests/unit/serialization/size_gatherer_container.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            new hpx::actions::transfer_action<test_action>(
+                hpx::threads::thread_priority_normal, buffer));
+    }
+
+    outp.set_parcel_id(hpx::parcelset::parcel::generate_unique_id());
+    outp.set_source(here);
+
+    std::vector<hpx::serialization::serialization_chunk>* chunks = 0;
+    if (!(out_archive_flags | hpx::serialization::disable_data_chunking))
+        chunks = new std::vector<hpx::serialization::serialization_chunk>();
```
Comment (sithhell) [inline tests/unit/serialization/size_gatherer_container.cpp]: new is called without matching delete.

Comment (hkaiser) [inline tests/unit/serialization/size_gatherer_container.cpp]: good catch

**Comment by hkaiser** [comment]
> How are zero copy chunks handled? Do potential zero copy arguments contribute to the size?

That should be fine now.

**Comment by biddisco** [comment]
This appears to be working well and is probably safe to merge - but I have not run the full set of tests.

**Comment by sithhell** [comment]
LGTM

**Comment by AntonBikineev** [comment]
LGTM

**Comment by sithhell** [comment]
Just for completeness sake: there is a minor performance regression with zero copy parcels. The overall benefit of that PR outweights those though.

**Comment by hkaiser** [comment]
> Just for completeness sake: there is a minor performance regression with zero copy parcels. The overall benefit of that PR outweights those though.

Have you done the measurements?

**Comment by sithhell** [comment]
On Sunday, May 31, 2015 13:59:42 Hartmut Kaiser wrote:
> > Just for completeness sake: there is a minor performance regression with
> > zero copy parcels. The overall benefit of that PR outweights those
> > though.
> Have you done the measurements?

Yes.

**Pull request merged**